GopherCAN.c

```diff
@@ -382,17 +388,36 @@ S8 send_parameter(CAN_INFO_STRUCT* param)
                 CAN_INFO_STRUCT* param = (CAN_INFO_STRUCT*) PARAMETERS[id];
                 err = encode_parameter(param, message.data, param_start, param_length);
                 if (err) return err;
-                else {
-                    param->last_tx = HAL_GetTick();
-                }
             }
 
             param_start = i;
             param_length = 1;
         }
     }
 
-    return tx_can_message(&message);
+    err = tx_can_message(&message);
+    if (err) return err;
+
+    // revisit parameters to update last tx
+    for (U8 i = 1; i < CAN_DATA_BYTES; i++) {
+        if (group->slots[i] != group->slots[i-1]) {
+            GCAN_PARAM_ID id = group->slots[i-1];
+            if (id < EMPTY_ID || id >= NUM_OF_PARAMETERS) {
+                return BAD_PARAMETER_ID;
+            } else {
+                CAN_INFO_STRUCT* param = (CAN_INFO_STRUCT*) PARAMETERS[id];
+                param->last_tx = HAL_GetTick();
+            }
+        }
+    }
+
+    // TODO if two parameter that are in the same group are sent back
```

Comment:
This is just suggesting putting a rate limit on sending a group, correct? Should that be defined in GopherCAN_config?

Reply:
Looks like this was already implemented correctly